### PR TITLE
Add a better knife spear

### PR DESCRIPTION
**Pull request by I-am-Erk**
#### Summary
```SUMMARY: Content "Adds a superior knife spear"```

#### Purpose of change
Knife spears as they exist are crude, and easily broken, because it's basically a knife duct taped to a stick.

That's a great starter weapon to get you out of reach of zombies right away, but it is possible to make a better quality spear using a knife.
image This is a survival knife version, but any knife that bolts to a tang could be taken off the tang, slotted and bolted to a split stick, and the split reinforced 

#### Describe the solution
Added a superior knife spear, which lacks the `fragile_melee` tag, and requires more skill and tools to make. It currently uses nails to attach the tang to the stick, but I would like to change it to bolts eventually, or better, have bolts or wooden dowels as options.

This led to removing the spike from the list of possible ingredients in a knife spear, because it isn't appropriate for superior spears and I want it to be possible to reinforce a knife spear to get a superior quality one. That's okay, because spikes as listed are pretty awful weapons: I added a spike spear, which uses spikes and lower-quality knives to make a really low end polearm, roughly the same as a fire hardened wood spear in damage but more fragile because it's just a hastily tied-on spike on a stick.

I also expanded the list of knives that can be used to make a knife spear.

#### Describe alternatives you've considered
I considered requiring a bit of wire/scrap metal as well but decided it doesn't add much realism.

#### Additional context
Also mildly weakened the existing knife spear, to open some room for the reinforced one to be more damaging due to the better leverage

### CODE REVIEW

**Comment by Mecares** [comment]
Maybe call the old knife spear instead spiked stick since it does not even use a knife to craft and is only a spike and a stick  hastily assembled together?

**Comment by I-am-Erk** [comment]
The old knife spear uses a knife now as well, if you have one.

**Comment by EirenexTheDragon** [comment]
the damage seems a little to high, when compared to a copper spear, which has a proper spearhead, having the same 16 damage. maybe a reduction to 1 damage?

**Comment by Maddremor** [comment]
> 
> 
> the damage seems a little to high, when compared to a copper spear, which has a proper spearhead, having the same 16 damage. maybe a reduction to do instead 15 damage?

While the copper spear has a purpose made spearhead, it's still made from copper by a survivor. The blades used to make the knife spear are made of steel with modern manufactoring methods, and should thus be more usable even if they are not designed as spearheads.

File: data/json/recipes/recipe_weapon.json [hunk 1]
```diff
@@ -316,16 +316,81 @@
...
+        [ "knife_rm42", 1 ],
+        [ "knife_trench", 1 ],
+        [ "tanto", 1 ],
+        [ "kirpan", 1 ]
+      ],
+      [ [ "nail", 2 ] ],
+      [ [ "cordage", 2, "LIST" ], [ "duct_tape", 100 ] ]
+    ]
+  },
+  {
```
Comment (Maddremor) [inline data/json/recipes/recipe_weapon.json]: The primitive recipie allows a crude spike to be used while the superior one don't. Upgrading simple to superior bypasses this.

Comment (I-am-Erk) [reply data/json/recipes/recipe_weapon.json]: I didn't really want to remove the ability to make an emergency spear with a spike, but I don't think it's appropriate for a superior spear. Found a workaround by making an even crappier knife spear option.

**Comment by I-am-Erk** [comment]
Exactly: copper isn't that great of a spear head material. The superior knife spear should actually be a pretty good weapon option, and it would definitely be better than hand-melted and sharpened copper.

**Comment by l29ah** [comment]
Copper spear is really weak for the stuff it requires, and i find it inferior even to a wooden spear, since it takes a lot more time to swing.

**Comment by kevingranade** [comment]
That's expected, the only time that a copper spear is a remotely reasonable option is when you're playing an innawoods character that doesn't have access to pre-made metal blades.  
In other words it's only an upgrade to wood or stone spears its inferior to everything else.

**Comment by TechyBen** [comment]
> That's expected, the only time that a copper spear is a remotely reasonable option is when you're playing an innawoods character that doesn't have access to pre-made metal blades.
> In other words it's only an upgrade to wood or stone spears its inferior to everything else.
Nice to see some alternatives for mechanics (crafting/survival vs looting/npc camp tools), not just stats (weak vs strong). Great stuff.

**Pull request merged**